core/test/matrix/permutation.cpp

```diff
@@ -100,6 +100,15 @@ TEST_F(Permutation, ReturnsNullValuesArrayWhenEmpty)
 }
 
 
+TEST_F(Permutation, FactorySetsCorrectPermuteMask)
```

Comment:
Now that the multiprecision testing is merged, this will need to slightly change. You only have to do it for these tests here, and you pretty much only have to do like what was there before. Hopefully it'll be rather fast.

As a measure of success, you can use the coverage report, if the coverage doesn't drop significantly then that means the templated tests are correct.